src/vsr/replica.zig

```diff
@@ -1031,6 +1065,22 @@ pub fn ReplicaType(
                 return;
             }
 
+            assert(self.status == .normal);
+            assert(message.header.view == self.view);
+            assert(self.primary() or self.backup());
+            assert(message.header.replica == self.primary_index(message.header.view));
+            assert(message.header.op > self.op_checkpoint());
+            assert(message.header.op > self.op);
+            assert(message.header.op > self.commit_min);
+
+            defer {
+                if (self.backup()) {
+                    // A prepare may already be committed if requested by repair() so take the max:
+                    self.commit_journal(std.math.max(message.header.commit, self.commit_max));
```

Comment:
```suggestion
                    self.commit_journal(message.header.commit);
```

I might be missing something, but it seems `self.commit_journal` already have the max logic internally?